index.js

```diff
@@ -244,7 +247,109 @@ class BitbucketScm extends Scm {
     }
 
     /**
-    * Retreive stats for the scm
+     * Get a commit sha for a specific repo#branch
+     * @method getCommitSha
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.scmUri     The scmUri to get commit sha of
+     * @param  {String}   config.token      The token used to authenticate to the SCM
+     * @return {Promise}                    Resolves to the sha for the scmUri
+     */
+    _getCommitSha(config) {
+        const scm = getScmUriParts(config.scmUri);
+        const branchUrl =
+            `${REPO_URL}/${scm.repoId}/refs/branches/${scm.branch}?access_key=${config.token}`;
+        const options = {
+            url: branchUrl,
+            method: 'GET'
+        };
+
+        return this.breaker.runCommand(options)
+            .then((response) => {
+                if (response.statusCode !== 200) {
+                    throw new Error(`STATUS CODE ${response.statusCode}: ${response.body}`);
+                }
+
+                return response.body.target.hash;
+            });
+    }
+
+    /**
+     * Fetch content of a file from Bitbucket
+     * @method getFile
+     * @param  {Object}   config              Configuration
+     * @param  {String}   config.scmUri       The scmUri to get permissions on
+     * @param  {String}   config.path         The file in the repo to fetch
+     * @param  {String}   config.token        The token used to authenticate to the SCM
+     * @param  {String}   config.ref          The reference to the SCM, either branch or sha
+     * @return {Promise}                      Resolves to the content of the file
+     */
+    _getFile(config) {
+        const scm = getScmUriParts(config.scmUri);
+        const fileUrl = `${API_URL_V1}/repositories/${scm.repoId}` +
```

Comment:
I understand you broke this up because of the line being too long. I'm wondering if you can still take advantage of string interpolation and keeping the linter happy. Something along the lines of

```
const urlRoot = `${API_URL_V1}/repositories/${scm.repoId}`
const urlSuffix = `src/${config.ref}/${config.path}`;
const urlParameters = `access_key=${config.token}`;

const fileUrl = `${urlRoot}/${urlSuffix}?${urlParameters}`;
```